src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -762,6 +762,90 @@ public static void Contains_NullValue_ThrowsArgumentNullException()
+        [InlineData("Hello", "Larger Hello", StringComparison.InvariantCultureIgnoreCase, false)]
+        [InlineData("Hello", "Goodbye", StringComparison.InvariantCultureIgnoreCase, false)]
+        [InlineData("", "", StringComparison.InvariantCultureIgnoreCase, true)]
+        [InlineData("", "hello", StringComparison.InvariantCultureIgnoreCase, false)]
+        [InlineData("Hello", "", StringComparison.InvariantCultureIgnoreCase, true)]
+        [InlineData("Hello", "ell" + SoftHyphen, StringComparison.InvariantCultureIgnoreCase, true)]
+        [InlineData("Hello", "Ell" + SoftHyphen, StringComparison.InvariantCultureIgnoreCase, true)]
+        // Ordinal
+        [InlineData("Hello", "ello", StringComparison.Ordinal, true)]
+        [InlineData("Hello", "ELL", StringComparison.Ordinal, false)]
+        [InlineData("Hello", "ElLo", StringComparison.Ordinal, false)]
+        [InlineData("Hello", "Larger Hello", StringComparison.Ordinal, false)]
+        [InlineData("Hello", "Goodbye", StringComparison.Ordinal, false)]
+        [InlineData("", "", StringComparison.Ordinal, true)]
+        [InlineData("", "hello", StringComparison.Ordinal, false)]
+        [InlineData("Hello", "", StringComparison.Ordinal, true)]
+        [InlineData("Hello", "ell" + SoftHyphen, StringComparison.Ordinal, false)]
+        [InlineData("Hello", "Ell" + SoftHyphen, StringComparison.Ordinal, false)]
+        // OrdinalIgnoreCase
+        [InlineData("Hello", "ello", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "ELL", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "ElLo", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "Larger Hello", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("Hello", "Goodbye", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("", "", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("", "hello", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("Hello", "", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "ell" + SoftHyphen, StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("Hello", "Ell" + SoftHyphen, StringComparison.OrdinalIgnoreCase, false)]
+        public static void Contains(string s, string value, StringComparison comparisonType, bool expected)
+        {
+            Assert.Equal(expected, s.Contains(value, comparisonType));
+        }
+
+        [Theory]
+        [InlineData(StringComparison.CurrentCulture)]
+        [InlineData(StringComparison.CurrentCultureIgnoreCase)]
+        [InlineData(StringComparison.InvariantCulture)]
+        [InlineData(StringComparison.InvariantCultureIgnoreCase)]
+        [InlineData(StringComparison.Ordinal)]
+        [InlineData(StringComparison.OrdinalIgnoreCase)]
+        public static void Contains_NullValue_ThrowsArgumentNullException(StringComparison comparisonType)
```

Comment:
Can we also get some tests for invalid `comparisonType` like the ones here: https://github.com/yvanin/corefx/blob/81ddecc9d6e8d7631568d5cf92ad97e710fdf3d1/src/System.Runtime/tests/System/StringTests.cs#L1156-L1164